lib/NewModel.ts

```diff
@@ -0,0 +1,284 @@
+        return Q.reject(err);
+      }
+    };
+
+    NewModel.waitForActive = function (timeout, next) {
+      return table.waitForActive(timeout, next);
+    };
+
+    NewModel.getTableReq = function () {
+      return table.buildTableReq(table.name, table.schema);
+    };
+
+    NewModel.conditionCheck = function (key, conditionCheckOptions, next) {
+      try {
+        return Model.conditionCheck(NewModel, key, conditionCheckOptions, next);
+      } catch (err) {
+        sendErrorToCallback(err, conditionCheckOptions, next);
+        return Q.reject(err);
+      }
+    };
+
+    function createTransactionFunction (func, val, optionsIndex) {
+      return async function (...args) {
+        if (typeof args[args.length - 1] === 'function') {
+          console.warn('Dynamoose Warning: Passing callback function into transaction method not allowed. Removing callback function from list of arguments.');
+          // Callback function passed in which is not allowed, removing that from arguments list
+          args.pop();
+        }
+
+        // Adding returnRequest to options
+        if (args.length >= optionsIndex + 1) {
+          args[optionsIndex] = {...args[optionsIndex], 'returnRequest': true};
+        } else if (args.length < optionsIndex) {
+          args[optionsIndex] = {'returnRequest': true};
+        } else {
+          args.push({'returnRequest': true});
+        }
+
+        const requestObj = await func(...args);
+
+        if (val === 'Update') {
+          delete requestObj.ReturnValues;
+        }
+
+        return {[val]: requestObj};
+      };
+    }
+    NewModel.transaction = [
+      // method - the method to be called when the user calls this method
+      // key - the key to be attached to the returned request object to pass into the DynamoDB transaciton request
+      // name - the name to attach to the resulting object for NewModel.transaction
+      // optionsIndex - the zero-based index for where the options parameter should be in the method arguments list
+      {'method': NewModel.get, 'key': 'Get', 'name': 'get', 'optionsIndex': 1},
+      {'method': NewModel.delete, 'key': 'Delete', 'name': 'delete', 'optionsIndex': 1},
+      {'method': NewModel.create, 'key': 'Put', 'name': 'create', 'optionsIndex': 1},
+      {'method': NewModel.update, 'key': 'Update', 'name': 'update', 'optionsIndex': 2},
+      {'method': NewModel.conditionCheck, 'key': 'ConditionCheck', 'name': 'conditionCheck', 'optionsIndex': 1}
+    ].reduce((original, newItem) => {
+      original[newItem.name] = createTransactionFunction(newItem.method, newItem.key, newItem.optionsIndex);
+      return original;
+    }, {});
+
+    // apply methods and statics
+    applyMethods(NewModel, schema);
+    applyStatics(NewModel, schema);
+
+    // set up middleware
+    for (const k in hooks) {
+      NewModel[k] = hooks[k];
+    }
+
+    table.init((err) => {
+      if (err) {
+        throw err;
+      }
+    });
+
+    return NewModel;
+  }
```

Comment:
New line at end of files.

```suggestion
  }
  
```

Reply:
Also, why is this indented? Are we missing a closing `}` somewhere?

Reply:
it's not reformatted completely yet